File: Python/Assignment2/Assignment2.py
import Shapes3D, sys

class Solver:
    shapesList = []
    total = 0
    with open(sys.argv[1]) as readFile:
        for line in readFile.readlines():
            lineData = line.split(',')   
            
            if lineData[0] == "cuboid":
                insertedShape = Shapes3D.Cuboid(
                    width = float(lineData[1]),
                    height = float(lineData[2]),
                    depth = float(lineData[3])
                )
                shapesList.append(insertedShape)

            elif lineData[0] == "cube":
                insertedShape = Shapes3D.Cube(
                    sidelength = float(lineData[1])
                )
                shapesList.append(insertedShape)

            elif lineData[0] == "cylinder":
                insertedShape = Shapes3D.Cylinder(
                    radius = float(lineData[1]),
                    height = float(lineData[2])
                )
                shapesList.append(insertedShape)

            elif lineData[0] == "sphere":
                insertedShape = Shapes3D.Sphere(
                    radius = float(lineData[1])
                )
                shapesList.append(insertedShape)

            elif lineData[0] == "prism":
                insertedShape = Shapes3D.Prism(
                    sidelength = float(lineData[1]),
                    faces = float(lineData[2]),
                    height = float(lineData[3])
                )
                shapesList.append(insertedShape)

            elif lineData[0] == "area":
                multiplier = float(lineData[1])
                for shape in shapesList:
                    total += shape.getSurfaceArea() * multiplier
                shapesList.clear()

            elif lineData[0] == "volume":
                multiplier = float(lineData[1])
                for shape in shapesList:
                    total += shape.getVolume() * multiplier
                shapesList.clear()

result = Solver()
print(result.total)

File: CSharp/ParsingShapes/Assignment2.cs
using David.Shapes3D;

namespace David.Assignment2 {

    public static class Solver {

            public static double total = 0;
            public static void ReadFile(string[] args) {
            
            foreach (string line in System.IO.File.ReadLines(args[0])) {
                string[] lineData = line.Split(',');

                Shape3D insertedShape;

                switch (lineData[0]) {
                    case "cuboid":
                        insertedShape = new Cuboid(
                            Convert.ToDouble(lineData[2]),
                            Convert.ToDouble(lineData[3]),
                            Convert.ToDouble(lineData[4])
                        );
                    break;

                    case "cube":
                        insertedShape = new Cube(
                            Convert.ToDouble(lineData[2])
                        );
                    break;
                    
                    case "cylinder":
                        insertedShape = new Cylinder(
                            Convert.ToDouble(lineData[2]),
                            Convert.ToDouble(lineData[3])
                        );
                    break;

                    case "sphere":
                            insertedShape = new Sphere(
                            Convert.ToDouble(lineData[2])
                        );
                    break;

                    default:
                        throw new FormatException("Please enter a valid shape!");
                }
                if (lineData[1] == "area") {
                    total += insertedShape.GetSurfaceArea();
                } else if (lineData[1] == "volume") {
                    total += insertedShape.GetVolume();
                }
            }
        }
    }
}